### ISSUE TITLE
"Stream" option disabled

### ISSUE THREAD

**Issue by sealweb** [gpodder/gpodder#199]
I am using gPodder on Ubuntu Gnome 16.04 LTS.

For all my subscribed Podcasts (including all recommended ones) the right-click option "stream" is disabled.  I am not sure if I am missing a dependency that makes this feature work, or if the feature is redundant. 

Thanks for this software,  apart from this issue it's really great!

**Comment by thp** [comment]
You need to pick a player that's not the default application in the preferences for "Audio player" and/or "Video player" (depending which file you want to stream).

The reason for this is that if you choose "default application", and you would "Stream" the podcast (which is a HTTP URL), it would open your web browser. This is why we disable the "Stream" option in that case. If you pick e.g. VLC or something, the option will be enabled, and VLC will be launched with the media URL, and it will stream the file.

**Comment by sealweb** [comment]
Excellent, thank you very much :-)

**Comment by noChamp** [comment]
hi
thanks for the great app.
I would like to understand why would you open a web browser when hitting 'stream' instead of the default audio app? whats the logic behind it?
thanks
M

**Comment by thp** [comment]
Short answer: Because the "default app" for http:// and https:// URLs is the browser.

Long answer:
The thing is, with the "default" app set, we don't know what it is - it's the system default app, and it can be different for each file type. So it's the same as double-clicking the file in the file browser, etc... - This works great for local files, since the system can figure out (based on the extension or file contents) what the file is, and pick the right application, but for streaming, we only have an URL (http:// or https://). Asking the system to open a URL will always result in the browser opening the file.

Of course, we could add some magic for e.g. figuring out the extension (in the case of Windows) or the mime type (in the case of Linux) for a given URL, but it won't work in all cases (e.g. if the download URL is http://example.com/download.php?episode=123, what file type is it?). Then again, one could do an HTTP HEAD request or something to figure out the mime type, etc.. but it's quite complicated. So yes, in theory it can be done with some more engineering/implementation, but hasn't been done yet.

But now that you mentioned it, if the RSS feed provides a proper mime type, we could from that one figure out the default application associated with that file type, and then assume that the application can handle http:// and https:// URLs and open that instead. Still, it requires going through the registry on Windows and through XDG files on Linux and something else on macOS to figure out what the application associated with a mime type / extension is.

In short: It's not impossible, but it takes some effort to develop, test and get right, and nobody has stepped up to do it yet.

**Comment by noChamp** [comment]
wow ! I am very impressed with the quick and details response !! awesome :) thanks . thumbs up !

**Comment by mlncn** [comment]
Thank you for the explanation!

This could use more prominent documentation... and perhaps recommended FLOSS applications for streaming?
